client/src/pages/ChatsBlock.jsx

```diff
@@ -0,0 +1,297 @@
+        if (error.message) {
+          setAnyError(error.message);
+        }
+
+        if (!error.response?.data) {
+          return;
+        }
+
+        if (error.response?.data.error) {
+          setAnyError(error.response?.data.error);
+        }
+
+        const { message } = error.response.data;
+
+        if (message) {
+          setAnyError(message);
+        }
+      });
+  }
+
+  function handleDeleteButton(chat) {
+    userService
+      .deleteRoom(chat.id)
+      .then(setAllChats)
+      .catch((error) => {
+        if (error.message) {
+          setAnyError(error.message);
+        }
+
+        if (!error.response?.data) {
+          return;
+        }
+
+        if (error.response?.data.error) {
+          setAnyError(error.response?.data.error);
+        }
+
+        const { message } = error.response.data;
+
+        if (message) {
+          setAnyError(message);
+        }
+      });
+  }
+
+  return (
+    <div className='w-full flex flex-col'>
+      <div className='flex justify-between mb-3'>
+        <div className='flex gap-6 items-center'>
+          <h3>Available chats</h3>
+        </div>
+
+        <div>
+          <Formik
+            initialValues={{ newChatName: '' }}
+            validateOnMount={true}
+            onSubmit={({ newChatName }, formikHelpers) => {
+              formikHelpers.setSubmitting(true);
+              formikHelpers.resetForm();
+
+              userService
+                .createNewChat(newChatName, user)
+                .then((res) => {
+                  setAllChats(res);
+                })
+                .catch((error) => {
+                  if (error.message) {
+                    setAnyError(error.message);
+                  }
+
+                  if (!error.response?.data) {
+                    return;
+                  }
+
+                  const { errors, message } = error.response.data;
+
+                  formikHelpers.setFieldError(
+                    'newChatName',
+                    errors?.newChatName
+                  );
+
+                  if (message) {
+                    setAnyError(message);
+                  }
+                })
+                .finally(() => formikHelpers.setSubmitting(false));
+            }}
+          >
+            {({ touched, errors, isSubmitting, resetForm, values }) => (
+              <Form className='flex items-center h-8'>
+                <div className='h-full'>
+                  <div>
+                    <Field
+                      className='h-8 rounded-xl p-2'
+                      validate={chatNameValidation}
+                      name='newChatName'
+                      type='text'
+                      id='newChatName'
+                      placeholder='Chat name'
+                      disabled={!user}
+                    />
+                  </div>
+
+                  {touched.newChatName &&
+                    errors.newChatName &&
+                    setAnyError(errors.newChatName)}
```

Comment:
Directly calling `setAnyError(errors.newChatName)` inside the JSX can lead to unnecessary re-renders. Consider handling this logic inside a `useEffect` or a separate function.